main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -1324,35 +1312,68 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
   }
 
   /**
-    * Performs naming on the given type parameters parameters `tparam0` under the given type environment `tenv0`.
+    * Performs naming on the given type parameters `tparam0` from the given cases `cases`.
     */
-  private def getTypeParams(tparams0: WeededAst.TypeParams, fparams0: List[WeededAst.FormalParam], tpe: WeededAst.Type, loc: SourceLocation)(implicit flix: Flix): Validation[List[NamedAst.TypeParam], NameError] = {
+  private def getTypeParamsFromCases(tparams0: WeededAst.TypeParams, cases: List[WeededAst.Case], loc: SourceLocation)(implicit flix: Flix): Validation[List[NamedAst.TypeParam], NameError] = {
     tparams0 match {
-      case WeededAst.TypeParams.Elided => getImplicitTypeParams(fparams0, tpe, loc)
-      case WeededAst.TypeParams.Explicit(tparams0) => getExplicitTypeParams(tparams0).toSuccess
+      case WeededAst.TypeParams.Elided => Nil.toSuccess // TODO allow implicit tparams?
```

Comment:
I see, so I guess ultimately we don't allow elided type parameters on enums. I think this is the right choice. Do we really want to see:

```
enum List {
  case Nil,
  case Cons(a, List[a])
}
```

that looks inconsistent to me.

Then I suggest that we rename the function `getTypeParamsFromCases` and remove the cases argument.

Reply:
Even when we don't allow elided tparams, we still need to take the cases as an argument to get the implicit tparams, in order to get the kinds for the explicit tparams.